ogr/ogrsf_frmts/mssqlspatial/ogrmssqlgeometrywriter.cpp

```diff
@@ -253,6 +253,11 @@ void OGRMSSQLGeometryWriter::WritePoint(double x, double y, double z, double m)
 void OGRMSSQLGeometryWriter::WriteSimpleCurve(OGRSimpleCurve *poGeom,
                                               int iStartIndex, int nCount)
 {
+    if (nColType == MSSQLCOLTYPE_GEOGRAPHY && poGeom->isClockwise())
+    {
+        poGeom->reversePoints();
```

Comment:
given that WriteSimpleCurve() can be called with iStartIndex != 0 (in the case of a compoundcurve),  reversePoints() can interfere with that. Probably the best would be to switch the starting and ending indices in the below loops and the increment step.
Furthermore doesn't the winding order depends on if the curve is an exterior one or an interior one ? and shouldn't that only be applied to curves that are rings of polygons/curvepolygons/multipolygons/multisurfaces ?

Reply:
acually isClockwise() must be only called on a full closed ring. if it is called on a part of a compoundcurve, its output is garbage (cf the "Assumes that the line is closed." comment in  OGRCurve::isClockwise() doc)

Reply:
Right, I'm looking athttps://learn.microsoft.com/en-us/sql/relational-databases/spatial/polygon?view=sql-server-ver16#orientation-of-spatial-data but while it mentions clearly the left-hand-rule for the outer ring(s) is doesn't say anything about the inner rings, should I check them too and apply the opposite rule (clockwise) ?

Reply:
> but while it mentions clearly the left-hand-rule for the outer ring(s) is doesn't say anything about the inner rings, should I check them too and apply the opposite rule (clockwise) ?

yes, I would expect the reverse convention to be used for inner rings